### PR TITLE
[FSR] Input width cleanup and yesNo field updtate

### LINKED ISSUE #60653: [FSR] Updated Text/Number components to use new width props
## Issue Description
Clean up existing components to use new [width props](https://design.va.gov/storybook/?path=/docs/components-va-text-input--default#widths) and remove unnecessary custom CSS classes. 

---
## Tasks
- [x] Update existing text and number components with new `width` prop
- [x] Remove extra custom CSS attached to existing components. 

## Acceptance Criteria
- [x] All text and number input components have a `width` prop attached to them 
- [x] All text and number input components have cleaned up classes
---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [x] **Attached to an Epic** (what body of work is this a part of?)
- [x] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [x] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [x] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

### PR DESCRIPTION
**Pull request by josephrlee**
## Summary

- Updated all `va-text-input` and `va-number-input` components to use the the `width` prop on the FSR
- Removed outdated `.input-size-*` CSS
- Changed `YesNo` form system pages to [use web component version](https://depo-platform-documentation.scrollhelp.site/developer-docs/va-forms-library-web-components-in-rjsf#VAFormsLibrary-WebcomponentsinRJSF-Howtouse).

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#60653

## Testing done

- Tested locally 

## What areas of the site does it impact?

*FSR*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/financial-status-report/components/household/DependentAges.jsx [hunk 1]
```diff
@@ -178,7 +178,7 @@ const DependentAges = ({
         name={`dependentAge-${i}`}
         onInput={({ target }) => handlers.updateDependents(target, i)}
         value={dependent.dependentAge}
-        className="input-size-2 no-wrap"
+        width="md"
         onBlur={event => handlers.handleBlur(event, i)}
```
Comment (amponce) [inline src/applications/financial-status-report/components/household/DependentAges.jsx:182]: We should double check mobile when this gets to stage

File: src/applications/financial-status-report/components/householdExpenses/InstallmentContract.jsx [hunk 2]
```diff
@@ -248,111 +248,108 @@ const InstallmentContract = props => {
...
+          inputmode="numeric"
+          label="Minimum monthly payment amount"
+          name="amountDueMonthly"
+          id="amountDueMonthly"
+          min={0}
+          max={MAXIMUM_INSTALLMENT_AMOUNT}
+          onInput={handleAmountDueMonthlyChange}
+          value={contractRecord.amountDueMonthly}
+          width="md"
+        />
```
Comment (amponce) [inline src/applications/financial-status-report/components/householdExpenses/InstallmentContract.jsx:318]: Missing uswds

File: src/applications/financial-status-report/pages/assets/recreationalVehicles/records.js [hunk 3]
```diff
@@ -32,9 +33,10 @@ const recreationalVehicleUISchema = {
     recVehicleAmount: {
       'ui:title':
         'What is the estimated value of all of your trailers, campers, and boats?',
+      'ui:webComponentField': VaTextInputField,
       'ui:options': {
         classNames: 'schemaform-currency-input',
-        widgetClassNames: 'input-size-4 vads-u-margin-bottom--3',
+        width: 'md',
```
Comment (amponce) [inline src/applications/financial-status-report/pages/assets/recreationalVehicles/records.js:39]: Did we verify that the validation still works on these new VaTextInput fields?

File: src/applications/financial-status-report/pages/income/spouse/benefits/records.js [hunk 4]
```diff
@@ -37,9 +36,10 @@ export const uiSchema = {
       education: {
         'ui:title':
           'How much does your spouse get each month for education benefits?',
+        'ui:webComponentField': VaTextInputField,
         'ui:options': {
           classNames: 'schemaform-currency-input',
-          widgetClassNames: 'input-size-3',
+          width: 'md',
```
Comment (amponce) [inline src/applications/financial-status-report/pages/income/spouse/benefits/records.js:42]: Another spot that I just want to confirm that validation didn't change with the new webComponentField

**Review by amponce — CHANGES_REQUESTED** [review]
Just a few small things to verify, and one update to the `uswds` prop and this will be good to go.

File: src/applications/financial-status-report/components/household/DependentAges.jsx (hunk 1 above)
Comment (josephrlee) [reply src/applications/financial-status-report/components/household/DependentAges.jsx:182]: Yup, I checked mobile. Using the `width="md"` prop fixes the no-wrap issues we were having.

File: src/applications/financial-status-report/pages/assets/recreationalVehicles/records.js (hunk 3 above)
Comment (josephrlee) [reply src/applications/financial-status-report/pages/assets/recreationalVehicles/records.js:39]: I believe so, I didn't see any errors.

Comment (josephrlee) [reply src/applications/financial-status-report/pages/assets/recreationalVehicles/records.js:39]: It throws an error if I leave it blank: Screenshot 2024-04-08 at 2 49 55 PM

File: src/applications/financial-status-report/pages/income/spouse/benefits/records.js (hunk 4 above)
Comment (josephrlee) [reply src/applications/financial-status-report/pages/income/spouse/benefits/records.js:42]: They work!

File: src/applications/financial-status-report/pages/assets/recreationalVehicles/records.js (hunk 3 above)
Comment (josephrlee) [reply src/applications/financial-status-report/pages/assets/recreationalVehicles/records.js:39]: They work!

**Review by amponce — APPROVED** [review]
Changes look good!

**Pull request merged**